README.adoc

```diff
@@ -96,6 +96,22 @@ We need to add ssh keys under authorized keys for all the nodes created by the i
 export SSH_PUBLIC_KEY="ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAklOUpkDHrfHY17SbrmTIpNLTGK9Tjom/BWDSUGPl+nafzlHDTYW7hdI4yZ5ew18JH4JW9jbhUFrviQzM7xlELEVf4h9lFX5QVkbPppSwg0cda3Pbv7kOdJ/MTyBlWXFCR+HAo3FXRitBqxiX1nKhXpHAZsMciLq8V6RjsNAQwdsdMFvSlVK/7XAt3FaoJoAsncM1Q9x5+3V0Ww68/eIFmb1zuUFljQJKprrX88XypNDvjYNby6vw/Pb0rwert/EnmZ+AW4OZPnTPI89ZPmVMLuayrD2cE86Z/il8b+gw3r3+1nKatmIkjn2so1d01QraTlMqVSsbxNrRFi9wrf+M7Q== schacon@mylaptop.local"
 ----
 
+=== MAILGUN_DOMAIN
+A domain is required for sending messages using the Mailgun API
+
+[source]
+----
+export MAILGUN_DOMAIN=your-mailgun-domain
+----
```

Comment:
```suggestion
----

The value of this environment variable must be base64 encoded.
```

Reply:
@alexeykazakov i changed the host_operator_secret.yaml to use `stringData` instead of `data`. I did this for two reasons 1) the user no longer has to worry about encoding the value - it can be passed in as a string and 2) since we are referencing an env var here rather than hard-coding the values it's also safe to use. also, the values will be encoded when the file is deployed. i tested this out by running the setup script for host cluster and looking at the secret after it was deployed on the host operator. you can read more about it here: https://kubernetes.io/docs/concepts/configuration/secret/

Reply:
If the mailgun api key is a plain string which doesn't require encoding then that's fine and we can use stringData. But sometimes keys contains characters which must be encoded (like ') then data must be used instead.

Reply:
hmm, from what i understood the string is encoded in the end, just not by us. i think i read that all stringData is merged into data ultimately. They showed some examples like: 

`apiVersion: v1
kind: Secret
metadata:
  name: mysecret
type: Opaque
data:
  username: YWRtaW4=
stringData:
  username: administrator`

which will result in: 
`apiVersion: v1
kind: Secret
metadata:
  creationTimestamp: 2018-11-15T20:46:46Z
  name: mysecret
  namespace: default
  resourceVersion: "7579"
  uid: 91460ecb-e917-11e8-98f2-025000000001
type: Opaque
data:
  username: YWRtaW5pc3RyYXRvcg==`

where YWRtaW5pc3RyYXRvcg== decodes to `administrator`

I guess we'll have to try it with the api key to know for sure though.

Reply:
That's correct. If we apply the following yaml:
```
apiVersion: v1
kind: Secret
metadata:
...
type: Opaque
stringData:
  username: admin
  key: opensesame
```
Then the following resource will be created instead:
```
...
type: Opaque
data:
  username: b3BlbnNlc2FtZQ==
  key: YWRtaW4=
```
But if `key` value is something like an RSA key:
```
-----BEGIN RSA PRIVATE KEY-----
MIICXAIBAAKBgQCqGKukO1De7zhZj6+H0qtjTkVxwTCpvKe4eCZ0FPqri0cb2JZfXJ/DgYSF6vUp
...
37sJ5QsW+sJyoNde3xH8vdXhzU7eT82D6X/scw9RZz+/6rCJ4p0=
-----END RSA PRIVATE KEY-----
```
then it's much easier to apply the encoded value using `data`:
```
apiVersion: v1
kind: Secret
metadata:
...
type: Opaque
data:
  username: b3BlbnNlc2FtZQ==
  key: LS0tLS...UgS0VZLS0tLS0=
```
So, kube has nothing to modify and will create a secret resource as is.
So, if we don't know for sure if the mailgun API keys should be encoded then it's safer to use `data` in our secret yaml.

Reply:
changed as part of: 090dd25b8925890865f394c183bb58bbf32c3507